### PR TITLE
feat: Add Entity.validate()

### PR DESCRIPTION
**Pull request by ntucker**
### Motivation

Easily customize validation - disable, enable in prod, etc.

### Solution

Refactor existing block and move into protected method. Call in normalize

### CODE REVIEW

**Comment by ntucker** [comment]
@skoging - the ultimate solution to validation code

File: packages/experimental/src/entity/Entity.ts [hunk 1]
```diff
@@ -112,6 +112,65 @@ export default abstract class Entity {
...
+    if (
+      visitedEntities[entityType][id].some((entity: any) => entity === input)
+    ) {
+      return id;
+    }
+    this.validate(processedEntity);
+    visitedEntities[entityType][id].push(input);
+
+    Object.keys(this.schema).forEach(key => {
+      if (Object.hasOwnProperty.call(processedEntity, key)) {
```
Comment (ljharb) [inline packages/experimental/src/entity/Entity.ts:156]:
```suggestion
      if (Object.prototype.hasOwnProperty.call(processedEntity, key)) {
```

File: packages/experimental/src/entity/Entity.ts [hunk 2]
```diff
@@ -207,60 +266,12 @@ First three members: ${JSON.stringify(processedEntity.slice(0, 3), null, 2)}`;
...
-          visitedEntities,
-        );
-      } else if (process.env.NODE_ENV !== 'production') {
-        if (!Object.hasOwnProperty.call(this.defaults, key)) {
-          const error = new Error(
-            `Schema key is missing in Entity
+    if (process.env.NODE_ENV !== 'production') {
+      Object.keys(this.schema).forEach(key => {
+        if (!Object.hasOwnProperty.call(processedEntity, key)) {
+          if (!Object.hasOwnProperty.call(this.defaults, key)) {
```
Comment (ljharb) [inline packages/experimental/src/entity/Entity.ts:272]:
```suggestion
        if (!Object.prototype.hasOwnProperty.call(processedEntity, key)) {
          if (!Object.prototype.hasOwnProperty.call(this.defaults, key)) {
```

File: packages/normalizr/src/entities/Entity.ts [hunk 3]
```diff
@@ -75,6 +75,77 @@ export default abstract class Entity extends SimpleRecord {
...
+    }
+    if (
+      visitedEntities[entityType][id].some((entity: any) => entity === input)
+    ) {
+      return id;
+    }
+    visitedEntities[entityType][id].push(input);
+
+    Object.keys(this.schema).forEach(key => {
+      if (Object.hasOwnProperty.call(processedEntity, key)) {
```
Comment (ljharb) [inline packages/normalizr/src/entities/Entity.ts:117]:
```suggestion
      if (Object.prototype.hasOwnProperty.call(processedEntity, key)) {
```

**Review by ljharb — APPROVED** [review]

**Pull request merged**